src/fsharp/ConstraintSolver.fs

```diff
@@ -1912,6 +1913,13 @@ and private SolveTypSubsumesTypWithReport (csenv:ConstraintSolverEnv) ndeep m tr
     TryD (fun () -> SolveTypSubsumesTypKeepAbbrevs csenv ndeep m trace ty1 ty2)
          (fun res -> ErrorD (ErrorsFromAddingSubsumptionConstraint(csenv.g,csenv.DisplayEnv,ty1,ty2,res,m)))
 
+and private SolveTypSubsumesTypWithSpecialReport (csenv:ConstraintSolverEnv) ndeep m trace ty1 ty2 = 
+    TryD (fun () -> SolveTypSubsumesTypKeepAbbrevs csenv ndeep m trace ty1 ty2)
+         (fun res ->
+            match SolveTypSubsumesTypKeepAbbrevs csenv ndeep m trace ty2 ty1 with // test if we can cast other way around
```

Comment:
I believe we want to undo the results of this new "solve" operation (regardless of whether it succeeds or not, using CollectThenUndo). We don't want the results of inference to be altered by giving better errors.